modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/transport/netty4/Netty4TransportMultiPortIntegrationIT.java

```diff
@@ -72,7 +72,7 @@ public void testThatTransportClientCanConnect() throws Exception {
             .build();
         // we have to test all the ports that the data node might be bound to
         try (TransportClient transportClient = new MockTransportClient(settings, Arrays.asList(Netty4Plugin.class))) {
-            for (int i = 0; i <= 10; i++) {
+            for (int i = 0; i <= NUMBER_OF_CLIENT_PORTS; i++) {
```

Comment:
I'm not that familiar with these parts, by I wonder why does this resolve the test failure? My understanding is that this only extends the list that the client can pick from. Adding some 300 ports there seems a bit odd to me. Are the `NoNodeAvailableException` in the failures caused by the port issue on Windows?

Reply:
It extends the list of ports we need to try to connect to. Windows blocks a lot of port ranges for some reason.